src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py

```diff
@@ -264,6 +265,21 @@ def from_neurodata_object(cls, neurodata_object: Container, dataset_name: Litera
         full_shape = get_data_shape(data=candidate_dataset)
         dtype = _infer_dtype(dataset=candidate_dataset)
 
+        if dtype == np.dtype(
```

Comment:
I would add this logic inside the last if in the branch below so it is a sub-case of object

Reply:
I mean here:
https://github.com/catalystneuro/neuroconv/blob/562bf7ff5222ea6d90f8ba6b4f440b3afcd64078/src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py#L295-L302